imageregistry/v1/types.go

```diff
@@ -306,6 +306,78 @@ type ImageRegistryConfigStorageIBMCOS struct {
+	PublicEndpoint EndpointAccessibility = "Public"
+
+	// ClearText is an AlibabaEncryptionMode. This means no encryption
+	ClearText AlibabaEncryptionMode = "ClearText"
+	// AES256 is an AlibabaEncryptionMode. This means AES256 encryption
+	AES256 AlibabaEncryptionMode = "AES256"
+	// KMS is an AlibabaEncryptionMode. This means KMS encryption
+	KMS AlibabaEncryptionMode = "KMS"
+)
+
+// EncryptionAlibaba this a union type in kube parlance.  Depending on the value for the encryptionType,
+// different pointers may be used
+type EncryptionAlibaba struct {
+	// EncryptionType defines the different encrytion types available
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default is `ClearText`.
+	// +kubebuilder:validation:Enum="ClearText"; "KMS"; "AES256"
+	// +optional
+	Type AlibabaEncryptionMode `json:"encryptionType"`
+
+	// KMS (key management service) is an encryption type provided to encrypt the provider
+	// +optional
+	KMS *KMSEncryptionAlibaba `json:"kms,omitempty"`
+}
+
+type KMSEncryptionAlibaba struct {
+	// KeyID holds the KMS encryption key ID
+	// +kubebuilder:validation:Required
+	KeyID string `json:"keyID"`
+}
+
+// ImageRegistryConfigStorageAlibabaOSS holds Alibaba Cloud OSS configuration.
+// Configures the registry to use Alibaba Cloud Object Storage Service for backend storage.
+// More about oss, you can look at the [official documentation](https://www.alibabacloud.com/help/product/31815.htm)
+type ImageRegistryConfigStorageAlibabaOSS struct {
+	// bucket is the bucket name in which you want to store the registry's data.
+	// About Bucket naming, more details you can look at the [official documentation](https://www.alibabacloud.com/help/doc-detail/257087.htm)
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default will be autogenerated in the form of <clusterid>-image-registry-<region>-<random string 27 chars>
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:MinLength=3
+	// +kubebuilder:validation:Pattern=`^[0-9a-z]+(-[0-9a-z]+)*$`
+	// +optional
+	Bucket string `json:"bucket,omitempty"`
+	// region is the Alibaba Cloud Region in which your bucket exists.
+	// For a list of regions, you can look at the [official documentation](https://www.alibabacloud.com/help/doc-detail/31837.html).
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default will be based on the installed Alibaba Cloud Region.
+	// +optional
+	Region string `json:"region,omitempty"`
+	// EndpointAccessibility specifies whether the registry use the OSS VPC internal endpoint
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default is `Internal`.
+	// +kubebuilder:validation:Enum="Internal"; "Public"
```

Comment:
is this syntax correct? The space especailly.

Reply:
Will remove the spaces.